Typos found in pandas

DimitriPapadopoulos opened the pull request:
See https://github.com/pandas-dev/pandas/pull/59665.

DimitriPapadopoulos commented:
Superseded by #3465.

musvaage commented:
codespell_lib/data/dictionary.txt

string `experientiallly` is erroneous

codespell_lib/data/dictionary_rare.txt

mangel is a type of beet fed to livestock

<hr>

_cf_:  `#59651` (pandas)

the cited pull for `salaraies` and `explicing` utilised these replacement strings

`salaries`
`explicitly`

DimitriPapadopoulos commented:
Thank you. I changed `experientiallly` to `experientially`.

Concerning `mangel`, I happen to happily eat mangel in the winter, but I thought it could also be a typo of `mangle`. You're right though, I don't see any occurrence of `mangel` as a typo of `mangle. It really is _mangel_, mostly in German text:
* https://github.com/search?q=mangel&type=code

However, `mangeled` is a typo of `mangled`.

I have added `salaraies` and `explicing`.

See ##3465, as I have merged multiple PRs.
